<?xml version="1.0" ?>
<robot name="robot" xmlns:xacro="http://www.ros.org/wiki/xacro">


  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.3 0.1 1"/>
  </material>
  <material name="blue">
    <color rgba="0.2 0.2 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 0.5"/>
  </material>

  <!-- Base link -->
  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>



  <!-- Joint 1 -->
  <joint name="joint1" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="link1"/>
    <origin rpy="0 0 0" xyz="0 0 0.1519"/>
  </joint>

  <!-- Joint 2 -->
  <joint name="joint2" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="link1"/>
    <child link="link2"/>
    <origin rpy="1.57075 0 0" xyz="0 0 0.24365"/>
  </joint>

  <!-- Joint 3 -->
  <joint name="joint3" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="link2"/>
    <child link="link3"/>
    <origin rpy="0 0 1.57075" xyz="0 0.21325 0"/>
  </joint>

  <!-- Joint 4 -->
  <joint name="joint4" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="link3"/>
    <child link="link4"/>
    <origin rpy="0 0 0" xyz="0.11235 0 0"/>
  </joint>

  <!-- Joint 5 -->
  <joint name="joint5" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="link4"/>
    <child link="link5"/>
    <origin rpy="1.57075 0 1.57075" xyz="0 0 -0.08535"/>
  </joint>

  <!-- Joint 6 -->
  <joint name="joint6" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="link5"/>
    <child link="link6"/>
    <origin rpy="1.57075 0 0" xyz="0 0 0.0819"/>
  </joint>



  <!-- Link 1 -->
  <link name="link1">
    <visual>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <!-- Link 2 -->
  <link name="link2">
    <visual>
      <origin rpy="0 0 1.57075" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <!-- Link 3 -->
  <link name="link3">
    <visual>
      <origin rpy="0 0 1.57075" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <!-- Link 4 -->
  <link name="link4">
    <visual>
      <origin rpy="0 0 1.57075" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <!-- Link 5 -->
  <link name="link5">
    <visual>
      <origin rpy="0 0 1.57075" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>

  <!-- Link 6 -->
  <link name="link6">
    <visual>
      <origin rpy="0 0 1.57075" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="black"/>
    </visual>
  </link>


</robot>
